File: cloudtimes-partner/src/main/java/com/cloudtimes/partner/weixin/ICtWeixinOfficialApiService.java
package com.cloudtimes.partner.weixin;

import com.cloudtimes.common.core.domain.AjaxResult;
import me.zhyd.oauth.model.AuthCallback;
import me.zhyd.oauth.model.AuthResponse;
import me.zhyd.oauth.model.AuthUser;

import java.security.NoSuchAlgorithmException;
import java.util.Map;

/**
 * 微公众号
 */
public interface ICtWeixinOfficialApiService {
    /**
     * 获取微信授权地址
     *
     * @return
     */
    public String getWXAuthURL();

    public AuthResponse<AuthUser> login(AuthCallback callback);

    public String getJSSDKSgin(String accessToken);


    public String getMPAccessToken();

    public AjaxResult getJSSDKSign(String accessToken,String ticket,String url) throws NoSuchAlgorithmException;
}

File: cloudtimes-partner/src/main/java/com/cloudtimes/partner/weixin/ICtWeixinOfficialApiService.kt
package com.cloudtimes.partner.weixin

import com.cloudtimes.common.core.domain.AjaxResult
import me.zhyd.oauth.model.AuthCallback
import me.zhyd.oauth.model.AuthResponse
import me.zhyd.oauth.model.AuthUser
import java.security.NoSuchAlgorithmException

/**
 * 微公众号
 */
interface ICtWeixinOfficialApiService {
    /**
     * 获取微信授权地址
     *
     * @return
     */
    fun getWXAuthURL(type: String?, inviteCode: String?): String

    fun login(callback: AuthCallback): AuthResponse<AuthUser>

    fun getJSSDKSign(accessToken: String): String

    fun getMPAccessToken(): String

    @Throws(NoSuchAlgorithmException::class)
    fun getJSSDKSign(accessToken: String, ticket: String, url: String): AjaxResult
}